compliance/collection/main.go

```diff
@@ -156,9 +157,13 @@ func manageSendToSensor(ctx context.Context, cli sensor.ComplianceService_Commun
 	}
 }
 
-func manageNodeScanLoop(ctx context.Context, i intervals.NodeScanIntervals, scanner nodeinventorizer.NodeInventorizer) <-chan *sensor.MsgFromCompliance {
+func manageNodeScanLoop(ctx context.Context, i intervals.NodeScanIntervals, scanner scannerV1.NodeInventoryServiceClient) <-chan *sensor.MsgFromCompliance {
 	sensorC := make(chan *sensor.MsgFromCompliance)
 	nodeName := getNode()
+	if scanner == nil {
+		log.Error("NodeInventoryServiceClient not provided. Node Scanning will be unavailable.")
+		return sensorC
+	}
```

Comment:
If we land in this branch, we need to close the chan.

```suggestion
	if scanner == nil {
		close(sensorC)
		log.Error("NodeInventoryServiceClient not provided. Node Scanning will be unavailable.")
		return sensorC
	}
```